roles/postgresql/tasks/main.yml

```diff
@@ -29,6 +21,76 @@
   register: "find_postgresql_data_path"
   tags: ["prepare", "prepare-postgresql"]
 
+- name: "Check if container {{ postgresql_container_name }} already exists on host"
+  community.docker.docker_container_info:
+    name: "{{ postgresql_container_name }}"
+  register: "postgresql_active_container"
+  tags: ["prepare", "prepare-postgresql"]
+
+- name: "Set postgresql_active_container_major_version"
+  ansible.builtin.set_fact:
+    postgresql_active_container_major_version: >-2
+      {{
+        postgresql_active_container.container.Config.Env
+        | select('search', '^PG_MAJOR=')
+        | first
+        | split('=')
+        | last
+      }}
+  when: "postgresql_active_container.exists"
+  tags: ["prepare", "prepare-postgresql"]
+
+- name: "Set postgresql_container_version to major version of existing container"
+  ansible.builtin.set_fact:
+    postgresql_container_version: "{{ postgresql_active_container_major_version }}"
+  when:
+    - "postgresql_active_container.exists"
+    - "not postgresql_container_custom_version"
+  tags: ["prepare", "prepare-postgresql"]
+
+- name: "Fail when installing postgresql_container_custom_version would require a database migration"
+  ansible.builtin.fail:
+    msg: |
+      postgresql_container_custom_version: {{ postgresql_container_custom_version }}
+      postgresql_active_container_major_version: {{ postgresql_active_container_major_version }}
+      postgresql_container_custom_version does not match the major version of the existing container.
+      PostgreSQL major version changes require a database migration, which this role does not provide.
+  when:
+    - "postgresql_active_container.exists"
+    - "postgresql_container_custom_version"
+    - "(postgresql_container_custom_version | int) != (postgresql_active_container_major_version | int)"
+  tags: ["prepare", "prepare-postgresql"]
+
+- name: "Set postgresql_container_version to postgresql_container_custom_version"
+  ansible.builtin.set_fact:
+    postgresql_container_version: "{{ postgresql_container_custom_version }}"
+  when: "postgresql_container_custom_version"
+  tags: ["prepare", "prepare-postgresql", "deploy", "deploy-postgresql"]
+
```

Comment:
I'd skip this whole logic tbh: If people want to use a specific version, they should migrate to it and then run the role again.